sale_loyalty_general_discount_promo_code/models/sale_order.py

```diff
@@ -0,0 +1,154 @@
+            and line.product_uom_qty
+            and line.price_total
+            and line.product_id.filtered_domain(reward._get_discount_product_domain())
+        )
+
+        return sum(
+            line.price_unit * line.product_uom_qty * (reward.discount / 100)
+            for line in lines_to_discount
+        )
+
+    def _get_total_saving_cheapest(self, reward):
+        self.ensure_one()
+        assert reward.discount_applicability == "cheapest"
+
+        cheapest_line = self._cheapest_line()
+        if not cheapest_line:
+            return 0
+        return (
+            cheapest_line.price_unit
+            * cheapest_line.product_uom_qty
+            * (reward.discount / 100)
+        )
+
+    def _get_total_saving_order(self, reward):
+        self.ensure_one()
+        assert reward.discount_applicability == "order"
+        total_saving = 0
+        for line in self.order_line:
+            if line.reward_id:
+                continue
+            total_saving += (
+                line.price_unit * line.product_uom_qty * (reward.discount / 100)
+            )
+        return total_saving
+
+    def _get_total_saved_amount(self, reward):
+        total_saving = 0
+        reward_applies_on = reward.discount_applicability
+        if reward_applies_on == "order":
+            total_saving = self._get_total_saving_order(reward)
+        elif reward_applies_on == "specific":
+            total_saving = self._get_total_saving_specific(reward)
+        elif reward_applies_on == "cheapest":
+            total_saving = self._get_total_saving_cheapest(reward)
+
+        return format_amount(self.env, total_saving, self.pricelist_id.currency_id)
+
+    def _get_reward_values_discount(self, reward, coupon, **kwargs):
+        rewards = super()._get_reward_values_discount(reward, coupon, **kwargs)
+
+        if (
+            not self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(
+                "sale_loyalty_general_discount_promo_code.allow_order_line_coupon_percentage"
+            )
+        ):
+            return rewards
+
+        for reward_line in rewards:
+            if (
+                reward.reward_type == "discount"
+                and reward.discount_mode == "percent"
+                and (
+                    reward.program_id.program_type
+                    in ["coupons", "promo_code", "promotion"]
+                )
+            ):
+                # Get the saved amount
+                # Cannot use the reward_line.price_unit for the saved amount
+                # because _get_reward_line_values is called twice.
+                # Once in _apply_program_reward and once in _update_programs_and_rewards
+                saved_amount = self._get_total_saved_amount(reward)
+                reward_line.update(
+                    {
+                        "display_type": "line_note",
+                        "name": _(f"You saved {saved_amount} with promo {coupon.code}"),
```

Comment:
don't you think it should be better to put the label of the promotion (e.g. `Code for 10% on orders - 10% on your order`)  instead of the actual code?

Reply:
the best name we can have is probably `loyalty.program.name`